pkg/ccl/schemachangerccl/sctestbackupccl/sctestbackupccl_test: TestBackupSuccess_multiregion_alter_table_alter_primary_key_rbr failed

pkg/ccl/schemachangerccl/sctestbackupccl/sctestbackupccl_test.TestBackupSuccess_multiregion_alter_table_alter_primary_key_rbr [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsIbmCloudLinuxS390x_UnitTestsS390x/20556293?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsIbmCloudLinuxS390x_UnitTestsS390x/20556293?buildTab=artifacts#/) on master @ [7635297adb8507d86e47e36c35a71d35e871e36f](https://github.com/cockroachdb/cockroach/commits/7635297adb8507d86e47e36c35a71d35e871e36f):


```
        	            	  | github.com/cockroachdb/cockroach/pkg/sql/schemachanger/sctest.waitForSchemaChangesToFinish
        	            	  | 	pkg/sql/schemachanger/sctest/framework.go:968
        	            	  | github.com/cockroachdb/cockroach/pkg/sql/schemachanger/sctest.exerciseBackupRestore.func1
        	            	  | 	pkg/sql/schemachanger/sctest/backup.go:587
        	            	  | testing.tRunner
        	            	  | 	GOROOT/src/testing/testing.go:1690
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_s390x.s:797
        	            	Wraps: (2) query 'SELECT status, job_type, description FROM [SHOW JOBS] WHERE job_type IN ('SCHEMA CHANGE', 'TYPEDESC SCHEMA CHANGE', 'NEW SCHEMA CHANGE') AND status NOT IN ('succeeded', 'failed')': expected:
        	            	  |
        	            	  | got:
        	            	  | running, NEW SCHEMA CHANGE, ALTER TABLE multiregion_db.public.table_regional_by_row ADD … NG HASH
        	            	Error types: (1) *withstack.withStack (2) *errutil.leafError
--- FAIL: TestBackupSuccess_multiregion_alter_table_alter_primary_key_rbr/post_commit_stage_5_of_24/restore_database (64.14s)
=== RUN   TestBackupSuccess_multiregion_alter_table_alter_primary_key_rbr/post_commit_stage_5_of_24/restore_database_with_schema_only
    sql_runner.go:117: 
        	Error Trace:	pkg/testutils/sqlutils/sql_runner.go:130
        	            				pkg/testutils/sqlutils/sql_runner.go:391
        	            				pkg/sql/schemachanger/sctest/framework.go:968
        	            				pkg/sql/schemachanger/sctest/backup.go:587
        	Error:      	Received unexpected error:
        	            	query 'SELECT status, job_type, description FROM [SHOW JOBS] WHERE job_type IN ('SCHEMA CHANGE', 'TYPEDESC SCHEMA CHANGE', 'NEW SCHEMA CHANGE') AND status NOT IN ('succeeded', 'failed')': expected:
        	            	(1) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/sqlutils.(*SQLRunner).CheckQueryResultsRetry.func1
        	            	  | 	pkg/testutils/sqlutils/sql_runner.go:394
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils.SucceedsWithinError.func1
        	            	  | 	pkg/testutils/soon.go:71
        	            	  | github.com/cockroachdb/cockroach/pkg/util/retry.ForDuration
        	            	  | 	pkg/util/retry/retry.go:346
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils.SucceedsWithinError
        	            	  | 	pkg/testutils/soon.go:77
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/sqlutils.(*SQLRunner).succeedsWithin
        	            	  | 	pkg/testutils/sqlutils/sql_runner.go:130
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/sqlutils.(*SQLRunner).CheckQueryResultsRetry
        	            	  | 	pkg/testutils/sqlutils/sql_runner.go:391
        	            	  | github.com/cockroachdb/cockroach/pkg/sql/schemachanger/sctest.waitForSchemaChangesToFinish
        	            	  | 	pkg/sql/schemachanger/sctest/framework.go:968
        	            	  | github.com/cockroachdb/cockroach/pkg/sql/schemachanger/sctest.exerciseBackupRestore.func1
        	            	  | 	pkg/sql/schemachanger/sctest/backup.go:587
        	            	  | testing.tRunner
        	            	  | 	GOROOT/src/testing/testing.go:1690
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_s390x.s:797
        	            	Wraps: (2) query 'SELECT status, job_type, description FROM [SHOW JOBS] WHERE job_type IN ('SCHEMA CHANGE', 'TYPEDESC SCHEMA CHANGE', 'NEW SCHEMA CHANGE') AND status NOT IN ('succeeded', 'failed')': expected:
        	            	  |
        	            	  | got:
        	            	  | running, NEW SCHEMA CHANGE, ALTER TABLE multiregion_db.public.table_regional_by_row ADD … NG HASH
        	            	Error types: (1) *withstack.withStack (2) *errutil.leafError
--- FAIL: TestBackupSuccess_multiregion_alter_table_alter_primary_key_rbr/post_commit_stage_5_of_24/restore_database_with_schema_only (54.42s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #154746 pkg/ccl/schemachangerccl/sctestbackupccl/sctestbackupccl_test: TestBackupSuccess_multiregion_alter_table_alter_primary_key_rbr failed [C-test-failure O-robot T-sql-foundations branch-release-25.4 release-blocker]
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestBackupSuccess_multiregion_alter_table_alter_primary_key_rbr.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-55119

pkg/ccl/schemachangerccl/sctestbackupccl/sctestbackupccl_test.TestBackupSuccess_multiregion_alter_table_alter_primary_key_rbr [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsIbmCloudLinuxS390x_UnitTestsS390x/20563085?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsIbmCloudLinuxS390x_UnitTestsS390x/20563085?buildTab=artifacts#/) on master @ [14ff833383127fa428b913c5281a8a6c862e2a41](https://github.com/cockroachdb/cockroach/commits/14ff833383127fa428b913c5281a8a6c862e2a41):


```
    datadriven.go:352: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/7602/execroot/com_github_cockroachdb_cockroach/bazel-out/s390x-fastbuild/bin/pkg/ccl/schemachangerccl/sctestbackupccl/sctestbackupccl_test_/sctestbackupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/schemachangerccl/testdata/end_to_end/alter_table_alter_primary_key_rbr/alter_table_alter_primary_key_rbr.definition:48: still running after 3m0.018201528s
    datadriven.go:352: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/7602/execroot/com_github_cockroachdb_cockroach/bazel-out/s390x-fastbuild/bin/pkg/ccl/schemachangerccl/sctestbackupccl/sctestbackupccl_test_/sctestbackupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/schemachangerccl/testdata/end_to_end/alter_table_alter_primary_key_rbr/alter_table_alter_primary_key_rbr.definition:48: still running after 3m10.018926395s
    datadriven.go:352: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/7602/execroot/com_github_cockroachdb_cockroach/bazel-out/s390x-fastbuild/bin/pkg/ccl/schemachangerccl/sctestbackupccl/sctestbackupccl_test_/sctestbackupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/schemachangerccl/testdata/end_to_end/alter_table_alter_primary_key_rbr/alter_table_alter_primary_key_rbr.definition:48: still running after 3m20.020037938s
    datadriven.go:352: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/7602/execroot/com_github_cockroachdb_cockroach/bazel-out/s390x-fastbuild/bin/pkg/ccl/schemachangerccl/sctestbackupccl/sctestbackupccl_test_/sctestbackupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/schemachangerccl/testdata/end_to_end/alter_table_alter_primary_key_rbr/alter_table_alter_primary_key_rbr.definition:48: still running after 3m30.020201432s
    datadriven.go:352: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/7602/execroot/com_github_cockroachdb_cockroach/bazel-out/s390x-fastbuild/bin/pkg/ccl/schemachangerccl/sctestbackupccl/sctestbackupccl_test_/sctestbackupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/schemachangerccl/testdata/end_to_end/alter_table_alter_primary_key_rbr/alter_table_alter_primary_key_rbr.definition:48: still running after 3m40.020364736s
    datadriven.go:352: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/7602/execroot/com_github_cockroachdb_cockroach/bazel-out/s390x-fastbuild/bin/pkg/ccl/schemachangerccl/sctestbackupccl/sctestbackupccl_test_/sctestbackupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/schemachangerccl/testdata/end_to_end/alter_table_alter_primary_key_rbr/alter_table_alter_primary_key_rbr.definition:48: still running after 3m50.020980036s
    datadriven.go:352: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/7602/execroot/com_github_cockroachdb_cockroach/bazel-out/s390x-fastbuild/bin/pkg/ccl/schemachangerccl/sctestbackupccl/sctestbackupccl_test_/sctestbackupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/schemachangerccl/testdata/end_to_end/alter_table_alter_primary_key_rbr/alter_table_alter_primary_key_rbr.definition:48: still running after 4m0.021476814s
    datadriven.go:352: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/7602/execroot/com_github_cockroachdb_cockroach/bazel-out/s390x-fastbuild/bin/pkg/ccl/schemachangerccl/sctestbackupccl/sctestbackupccl_test_/sctestbackupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/schemachangerccl/testdata/end_to_end/alter_table_alter_primary_key_rbr/alter_table_alter_primary_key_rbr.definition:48: still running after 4m10.021535918s
    datadriven.go:352: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/7602/execroot/com_github_cockroachdb_cockroach/bazel-out/s390x-fastbuild/bin/pkg/ccl/schemachangerccl/sctestbackupccl/sctestbackupccl_test_/sctestbackupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/schemachangerccl/testdata/end_to_end/alter_table_alter_primary_key_rbr/alter_table_alter_primary_key_rbr.definition:48: still running after 4m20.022501273s
    panic.go:635: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/322f18dcd157731c8bfb5905550c09b4/logTestBackupSuccess_multiregion_alter_table_alter_primary_key_rbr8354474
--- FAIL: TestBackupSuccess_multiregion_alter_table_alter_primary_key_rbr (261.76s)
=== RUN   TestBackupSuccess_multiregion_alter_table_alter_primary_key_rbr/post_commit_stage_2_of_24
--- FAIL: TestBackupSuccess_multiregion_alter_table_alter_primary_key_rbr/post_commit_stage_2_of_24 (169.11s)
=== RUN   TestBackupSuccess_multiregion_alter_table_alter_primary_key_rbr/post_commit_stage_2_of_24/restore_all_tables_in_database
    sql_runner.go:117: 
        	Error Trace:	pkg/testutils/sqlutils/sql_runner.go:130
        	            				pkg/testutils/sqlutils/sql_runner.go:391
        	            				pkg/sql/schemachanger/sctest/framework.go:968
        	            				pkg/sql/schemachanger/sctest/backup.go:587
        	Error:      	Received unexpected error:
        	            	query 'SELECT status, job_type, description FROM [SHOW JOBS] WHERE job_type IN ('SCHEMA CHANGE', 'TYPEDESC SCHEMA CHANGE', 'NEW SCHEMA CHANGE') AND status NOT IN ('succeeded', 'failed')': expected:
        	            	(1) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/sqlutils.(*SQLRunner).CheckQueryResultsRetry.func1
        	            	  | 	pkg/testutils/sqlutils/sql_runner.go:394
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils.SucceedsWithinError.func1
        	            	  | 	pkg/testutils/soon.go:71
        	            	  | github.com/cockroachdb/cockroach/pkg/util/retry.ForDuration
        	            	  | 	pkg/util/retry/retry.go:346
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils.SucceedsWithinError
        	            	  | 	pkg/testutils/soon.go:77
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/sqlutils.(*SQLRunner).succeedsWithin
        	            	  | 	pkg/testutils/sqlutils/sql_runner.go:130
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/sqlutils.(*SQLRunner).CheckQueryResultsRetry
        	            	  | 	pkg/testutils/sqlutils/sql_runner.go:391
        	            	  | github.com/cockroachdb/cockroach/pkg/sql/schemachanger/sctest.waitForSchemaChangesToFinish
        	            	  | 	pkg/sql/schemachanger/sctest/framework.go:968
        	            	  | github.com/cockroachdb/cockroach/pkg/sql/schemachanger/sctest.exerciseBackupRestore.func1
        	            	  | 	pkg/sql/schemachanger/sctest/backup.go:587
        	            	  | testing.tRunner
        	            	  | 	GOROOT/src/testing/testing.go:1690
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_s390x.s:797
        	            	Wraps: (2) query 'SELECT status, job_type, description FROM [SHOW JOBS] WHERE job_type IN ('SCHEMA CHANGE', 'TYPEDESC SCHEMA CHANGE', 'NEW SCHEMA CHANGE') AND status NOT IN ('succeeded', 'failed')': expected:
        	            	  |
        	            	  | got:
        	            	  | running, NEW SCHEMA CHANGE, ALTER TABLE multiregion_db.public.table_regional_by_row ADD … NG HASH
        	            	Error types: (1) *withstack.withStack (2) *errutil.leafError
--- FAIL: TestBackupSuccess_multiregion_alter_table_alter_primary_key_rbr/post_commit_stage_2_of_24/restore_all_tables_in_database (55.72s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #154746 pkg/ccl/schemachangerccl/sctestbackupccl/sctestbackupccl_test: TestBackupSuccess_multiregion_alter_table_alter_primary_key_rbr failed [C-test-failure O-robot T-sql-foundations branch-release-25.4 release-blocker]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestBackupSuccess_multiregion_alter_table_alter_primary_key_rbr.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

pkg/ccl/schemachangerccl/sctestbackupccl/sctestbackupccl_test.TestBackupSuccess_multiregion_alter_table_alter_primary_key_rbr [failed](https://mesolite.cluster.engflow.com/invocations/default/cd5e41f7-257b-413c-9b30-dbfd4617a94d?testReportRun=8&testReportShard=40&testReportAttempt=1#targets-Ly9wa2cvY2NsL3NjaGVtYWNoYW5nZXJjY2wvc2N0ZXN0YmFja3VwY2NsOnNjdGVzdGJhY2t1cGNjbF90ZXN0) on master @ [5cda21f8745d4584fa16c82b4415cdb4c82cc67f](https://github.com/cockroachdb/cockroach/commits/5cda21f8745d4584fa16c82b4415cdb4c82cc67f):


```
    datadriven.go:357: /mnt/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/schemachangerccl/sctestbackupccl/sctestbackupccl_test_/sctestbackupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/schemachangerccl/testdata/end_to_end/alter_table_alter_primary_key_rbr/alter_table_alter_primary_key_rbr.definition:58: still running after 2m40.006892508s
    datadriven.go:357: /mnt/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/schemachangerccl/sctestbackupccl/sctestbackupccl_test_/sctestbackupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/schemachangerccl/testdata/end_to_end/alter_table_alter_primary_key_rbr/alter_table_alter_primary_key_rbr.definition:58: still running after 2m50.007677606s
    datadriven.go:357: /mnt/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/schemachangerccl/sctestbackupccl/sctestbackupccl_test_/sctestbackupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/schemachangerccl/testdata/end_to_end/alter_table_alter_primary_key_rbr/alter_table_alter_primary_key_rbr.definition:58: still running after 3m0.007974549s
    datadriven.go:357: /mnt/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/schemachangerccl/sctestbackupccl/sctestbackupccl_test_/sctestbackupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/schemachangerccl/testdata/end_to_end/alter_table_alter_primary_key_rbr/alter_table_alter_primary_key_rbr.definition:58: still running after 3m10.00814821s
    datadriven.go:357: /mnt/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/schemachangerccl/sctestbackupccl/sctestbackupccl_test_/sctestbackupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/schemachangerccl/testdata/end_to_end/alter_table_alter_primary_key_rbr/alter_table_alter_primary_key_rbr.definition:58: still running after 3m20.008271204s
    datadriven.go:357: /mnt/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/schemachangerccl/sctestbackupccl/sctestbackupccl_test_/sctestbackupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/schemachangerccl/testdata/end_to_end/alter_table_alter_primary_key_rbr/alter_table_alter_primary_key_rbr.definition:58: still running after 3m30.009276711s
    datadriven.go:357: /mnt/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/schemachangerccl/sctestbackupccl/sctestbackupccl_test_/sctestbackupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/schemachangerccl/testdata/end_to_end/alter_table_alter_primary_key_rbr/alter_table_alter_primary_key_rbr.definition:58: still running after 3m40.010054415s
    datadriven.go:357: /mnt/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/schemachangerccl/sctestbackupccl/sctestbackupccl_test_/sctestbackupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/schemachangerccl/testdata/end_to_end/alter_table_alter_primary_key_rbr/alter_table_alter_primary_key_rbr.definition:58: still running after 3m50.010138221s
    datadriven.go:357: /mnt/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/schemachangerccl/sctestbackupccl/sctestbackupccl_test_/sctestbackupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/schemachangerccl/testdata/end_to_end/alter_table_alter_primary_key_rbr/alter_table_alter_primary_key_rbr.definition:58: still running after 4m0.010188646s
    panic.go:635: -- test log scope end --
test logs left over in: outputs.zip/logTestBackupSuccess_multiregion_alter_table_alter_primary_key_rbr2775252456
--- FAIL: TestBackupSuccess_multiregion_alter_table_alter_primary_key_rbr (241.76s)
=== RUN   TestBackupSuccess_multiregion_alter_table_alter_primary_key_rbr/post_commit_stage_7_of_24
--- FAIL: TestBackupSuccess_multiregion_alter_table_alter_primary_key_rbr/post_commit_stage_7_of_24 (154.33s)
=== RUN   TestBackupSuccess_multiregion_alter_table_alter_primary_key_rbr/post_commit_stage_7_of_24/restore_all_tables_in_database
    sql_runner.go:117: 
        	Error Trace:	pkg/testutils/sqlutils/sql_runner.go:130
        	            				pkg/testutils/sqlutils/sql_runner.go:391
        	            				pkg/sql/schemachanger/sctest/framework.go:968
        	            				pkg/sql/schemachanger/sctest/backup.go:587
        	Error:      	Received unexpected error:
        	            	query 'SELECT status, job_type, description FROM [SHOW JOBS] WHERE job_type IN ('SCHEMA CHANGE', 'TYPEDESC SCHEMA CHANGE', 'NEW SCHEMA CHANGE') AND status NOT IN ('succeeded', 'failed')': expected:
        	            	(1) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/sqlutils.(*SQLRunner).CheckQueryResultsRetry.func1
        	            	  | 	pkg/testutils/sqlutils/sql_runner.go:394
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils.SucceedsWithinError.func1
        	            	  | 	pkg/testutils/soon.go:71
        	            	  | github.com/cockroachdb/cockroach/pkg/util/retry.ForDuration
        	            	  | 	pkg/util/retry/retry.go:355
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils.SucceedsWithinError
        	            	  | 	pkg/testutils/soon.go:77
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/sqlutils.(*SQLRunner).succeedsWithin
        	            	  | 	pkg/testutils/sqlutils/sql_runner.go:130
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/sqlutils.(*SQLRunner).CheckQueryResultsRetry
        	            	  | 	pkg/testutils/sqlutils/sql_runner.go:391
        	            	  | github.com/cockroachdb/cockroach/pkg/sql/schemachanger/sctest.waitForSchemaChangesToFinish
        	            	  | 	pkg/sql/schemachanger/sctest/framework.go:968
        	            	  | github.com/cockroachdb/cockroach/pkg/sql/schemachanger/sctest.exerciseBackupRestore.func1
        	            	  | 	pkg/sql/schemachanger/sctest/backup.go:587
        	            	  | testing.tRunner
        	            	  | 	GOROOT/src/testing/testing.go:1690
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_amd64.s:1700
        	            	Wraps: (2) query 'SELECT status, job_type, description FROM [SHOW JOBS] WHERE job_type IN ('SCHEMA CHANGE', 'TYPEDESC SCHEMA CHANGE', 'NEW SCHEMA CHANGE') AND status NOT IN ('succeeded', 'failed')': expected:
        	            	  |
        	            	  | got:
        	            	  | running, NEW SCHEMA CHANGE, ALTER TABLE multiregion_db.public.table_regional_by_row ADD … NG HASH
        	            	Error types: (1) *withstack.withStack (2) *errutil.leafError
--- FAIL: TestBackupSuccess_multiregion_alter_table_alter_primary_key_rbr/post_commit_stage_7_of_24/restore_all_tables_in_database (55.91s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=8</code>
 - <code>shard=40</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestBackupSuccess_multiregion_alter_table_alter_primary_key_rbr.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

pkg/ccl/schemachangerccl/sctestbackupccl/sctestbackupccl_test.TestBackupSuccess_multiregion_alter_table_alter_primary_key_rbr [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/20601822?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/20601822?buildTab=artifacts#/) on master @ [d0b9c0832caaa1d309fd3712c9c8e42064be0dec](https://github.com/cockroachdb/cockroach/commits/d0b9c0832caaa1d309fd3712c9c8e42064be0dec):


```
    datadriven.go:357: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/18402/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/ccl/schemachangerccl/sctestbackupccl/sctestbackupccl_test_/sctestbackupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/schemachangerccl/testdata/end_to_end/alter_table_alter_primary_key_rbr/alter_table_alter_primary_key_rbr.definition:58: still running after 1m50.012841482s
    datadriven.go:357: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/18402/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/ccl/schemachangerccl/sctestbackupccl/sctestbackupccl_test_/sctestbackupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/schemachangerccl/testdata/end_to_end/alter_table_alter_primary_key_rbr/alter_table_alter_primary_key_rbr.definition:58: still running after 2m0.013727326s
    datadriven.go:357: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/18402/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/ccl/schemachangerccl/sctestbackupccl/sctestbackupccl_test_/sctestbackupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/schemachangerccl/testdata/end_to_end/alter_table_alter_primary_key_rbr/alter_table_alter_primary_key_rbr.definition:58: still running after 2m10.014584096s
    datadriven.go:357: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/18402/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/ccl/schemachangerccl/sctestbackupccl/sctestbackupccl_test_/sctestbackupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/schemachangerccl/testdata/end_to_end/alter_table_alter_primary_key_rbr/alter_table_alter_primary_key_rbr.definition:58: still running after 2m20.014809487s
    datadriven.go:357: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/18402/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/ccl/schemachangerccl/sctestbackupccl/sctestbackupccl_test_/sctestbackupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/schemachangerccl/testdata/end_to_end/alter_table_alter_primary_key_rbr/alter_table_alter_primary_key_rbr.definition:58: still running after 2m30.015904469s
    datadriven.go:357: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/18402/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/ccl/schemachangerccl/sctestbackupccl/sctestbackupccl_test_/sctestbackupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/schemachangerccl/testdata/end_to_end/alter_table_alter_primary_key_rbr/alter_table_alter_primary_key_rbr.definition:58: still running after 2m40.016159435s
    datadriven.go:357: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/18402/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/ccl/schemachangerccl/sctestbackupccl/sctestbackupccl_test_/sctestbackupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/schemachangerccl/testdata/end_to_end/alter_table_alter_primary_key_rbr/alter_table_alter_primary_key_rbr.definition:58: still running after 2m50.018758008s
    datadriven.go:357: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/18402/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/ccl/schemachangerccl/sctestbackupccl/sctestbackupccl_test_/sctestbackupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/schemachangerccl/testdata/end_to_end/alter_table_alter_primary_key_rbr/alter_table_alter_primary_key_rbr.definition:58: still running after 3m0.019069151s
    datadriven.go:357: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/18402/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/ccl/schemachangerccl/sctestbackupccl/sctestbackupccl_test_/sctestbackupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/schemachangerccl/testdata/end_to_end/alter_table_alter_primary_key_rbr/alter_table_alter_primary_key_rbr.definition:58: still running after 3m10.019764067s
    panic.go:635: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/dd705d6ab6b4d56be60a3b91368d4a06/logTestBackupSuccess_multiregion_alter_table_alter_primary_key_rbr3731522722
--- FAIL: TestBackupSuccess_multiregion_alter_table_alter_primary_key_rbr (191.84s)
=== RUN   TestBackupSuccess_multiregion_alter_table_alter_primary_key_rbr/post_commit_stage_4_of_24
--- FAIL: TestBackupSuccess_multiregion_alter_table_alter_primary_key_rbr/post_commit_stage_4_of_24 (122.92s)
=== RUN   TestBackupSuccess_multiregion_alter_table_alter_primary_key_rbr/post_commit_stage_4_of_24/restore_database
    sql_runner.go:117: 
        	Error Trace:	pkg/testutils/sqlutils/sql_runner.go:130
        	            				pkg/testutils/sqlutils/sql_runner.go:391
        	            				pkg/sql/schemachanger/sctest/framework.go:968
        	            				pkg/sql/schemachanger/sctest/backup.go:587
        	Error:      	Received unexpected error:
        	            	query 'SELECT status, job_type, description FROM [SHOW JOBS] WHERE job_type IN ('SCHEMA CHANGE', 'TYPEDESC SCHEMA CHANGE', 'NEW SCHEMA CHANGE') AND status NOT IN ('succeeded', 'failed')': expected:
        	            	(1) attached stack trace
        	            	  -- stack trace:
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/sqlutils.(*SQLRunner).CheckQueryResultsRetry.func1
        	            	  | 	pkg/testutils/sqlutils/sql_runner.go:394
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils.SucceedsWithinError.func1
        	            	  | 	pkg/testutils/soon.go:71
        	            	  | github.com/cockroachdb/cockroach/pkg/util/retry.ForDuration
        	            	  | 	pkg/util/retry/retry.go:355
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils.SucceedsWithinError
        	            	  | 	pkg/testutils/soon.go:77
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/sqlutils.(*SQLRunner).succeedsWithin
        	            	  | 	pkg/testutils/sqlutils/sql_runner.go:130
        	            	  | github.com/cockroachdb/cockroach/pkg/testutils/sqlutils.(*SQLRunner).CheckQueryResultsRetry
        	            	  | 	pkg/testutils/sqlutils/sql_runner.go:391
        	            	  | github.com/cockroachdb/cockroach/pkg/sql/schemachanger/sctest.waitForSchemaChangesToFinish
        	            	  | 	pkg/sql/schemachanger/sctest/framework.go:968
        	            	  | github.com/cockroachdb/cockroach/pkg/sql/schemachanger/sctest.exerciseBackupRestore.func1
        	            	  | 	pkg/sql/schemachanger/sctest/backup.go:587
        	            	  | testing.tRunner
        	            	  | 	GOROOT/src/testing/testing.go:1690
        	            	  | runtime.goexit
        	            	  | 	src/runtime/asm_arm64.s:1223
        	            	Wraps: (2) query 'SELECT status, job_type, description FROM [SHOW JOBS] WHERE job_type IN ('SCHEMA CHANGE', 'TYPEDESC SCHEMA CHANGE', 'NEW SCHEMA CHANGE') AND status NOT IN ('succeeded', 'failed')': expected:
        	            	  |
        	            	  | got:
        	            	  | running, NEW SCHEMA CHANGE, ALTER TABLE multiregion_db.public.table_regional_by_row ADD … NG HASH
        	            	Error types: (1) *withstack.withStack (2) *errutil.leafError
--- FAIL: TestBackupSuccess_multiregion_alter_table_alter_primary_key_rbr/post_commit_stage_4_of_24/restore_database (48.29s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #155137 pkg/ccl/schemachangerccl/sctestbackupccl/sctestbackupccl_test: TestBackupSuccess_multiregion_alter_table_alter_primary_key_rbr failed [C-test-failure O-robot P-2 T-sql-foundations branch-release-25.4]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestBackupSuccess_multiregion_alter_table_alter_primary_key_rbr.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/pkg/sql/schemachanger/sctest/backup.go
+++ b/pkg/sql/schemachanger/sctest/backup.go
@@ -129,7 +129,7 @@ const skipRate = .6
 
 // Stop once max stages has been hit.
 const maxStagesToTest = 8
-const maxStagesToTestMultiRegion = 6
+const maxStagesToTestMultiRegion = 4
 
 // shouldSkipBackup samples about 60% of total stages or up to
 // max stages.

--- a/pkg/sql/schemachanger/sctest/framework.go
+++ b/pkg/sql/schemachanger/sctest/framework.go
@@ -965,6 +965,14 @@ func waitForSchemaChangesToSucceed(t *testing.T, tdb *sqlutils.SQLRunner) {
 }
 
 func waitForSchemaChangesToFinish(t *testing.T, tdb *sqlutils.SQLRunner) {
+	// Wait longer on stress for schema changes.
+	if skip.Stress() {
+		old := tdb.SucceedsSoonDuration
+		tdb.SucceedsSoonDuration = time.Minute * 2
+		defer func() {
+			tdb.SucceedsSoonDuration = old
+		}()
+	}
 	tdb.CheckQueryResultsRetry(
 		t, schemaChangeWaitQuery(`('succeeded', 'failed')`), [][]string{},
 	)

